velox/connectors/hive/HivePartitionFunction.cpp

```diff
@@ -29,19 +30,68 @@ void hashTyped(
       TypeTraits<kind>::name);
 }
 
-template <>
-void hashTyped<TypeKind::BOOLEAN>(
+template <typename T>
+void abstractHashTyped(
     const DecodedVector& values,
     vector_size_t size,
     bool mix,
-    std::vector<uint32_t>& hashes) {
+    std::vector<uint32_t>& hashes,
+    std::function<uint32_t(const T&)> const& f) {
```

Comment:
naming: f -> hashOne or something similar
order of parameters: input, followed by in/out, followed by output 

It might be more efficient to templatize on the function type. It would be nice to add a benchmark in a follow-up PR.

```
template <typename T, typename THash>
void abstractHashTyped(
    const DecodedVector& values,
    vector_size_t size,
    bool mix,
    THash hashOne,
    std::vector<uint32_t>& hashes)
```

Reply:
> Should I open an issue to track the progress of benchmark?

Sure.

Reply:
I've created a [PR](https://github.com/facebookincubator/velox/pull/1633) adding the benchmark, thanks.